internal/persistence/ent/client/convert.go

```diff
@@ -69,3 +75,28 @@ func convertTask(task *db.Task) *datatypes.Task {
 		WorkflowID:  task.WorkflowID,
 	}
 }
+
+// convertUser converts an entgo `db.User` representation to a
+// `datatypes.User` representation.
+func convertUser(dbu *db.User) *datatypes.User {
+	u := datatypes.User{
+		UUID:      dbu.UUID,
+		CreatedAt: &dbu.CreatedAt, // CreatedAt is never null in the database.
+	}
+
+	// Set optional fields if they are not empty.
+	if dbu.Email != "" {
+		u.Email = &dbu.Email
+	}
+	if dbu.Name != "" {
+		u.Name = &dbu.Name
+	}
+	if dbu.JwtIss != "" {
+		u.JWTIss = &dbu.JwtIss
+	}
+	if dbu.JwtSub != "" {
+		u.JWTSub = &dbu.JwtSub
+	}
```

Comment:
If the zero value could be considered as nil, this transformation would be straight forward. Also, the other way around, I think there is a flag that allows Ent to return nil fields directly from optional columns, which would also simplify this transformation.